controller/internal/enforcer/applicationproxy/http/http.go

```diff
@@ -659,11 +659,14 @@ func userCredentials(serviceID string, r *http.Request, authorizer *auth.Process
 	}
 
 	userAttributes, redirect, refreshedToken, err := authorizer.DecodeUserClaims(serviceID, userToken, userCerts, r)
-	if err == nil && len(userAttributes) > 0 {
+	if len(userAttributes) > 0 {
 		userRecord := &collector.UserRecord{Claims: userAttributes}
 		c.CollectUserEvent(userRecord)
 		state.stats.Source.UserID = userRecord.ID
 		state.stats.Source.Type = collector.EndpointTypeClaims
```

Comment:
Can you verify this with OAUTH? I remember a combination giving me trouble.

Reply:
there is no return statement in the OAUTH Validate() which would produce `userAttributes` and an error. When it produces `userAttributes` err will always be nil, and when there is an error, `userAttributes` will always be empty.